Toolkit/ArcGISToolkit/AR/ArcGISARView.swift

```diff
@@ -0,0 +1,512 @@
+    @available(iOS 11.3, *)
+    public func sessionShouldAttemptRelocalization(_ session: ARSession) -> Bool {
+        return arSCNViewDelegate?.sessionShouldAttemptRelocalization?(session) ?? false
+    }
+    
+    public func session(_ session: ARSession, didOutputAudioSampleBuffer audioSampleBuffer: CMSampleBuffer) {
+        arSCNViewDelegate?.session?(session, didOutputAudioSampleBuffer: audioSampleBuffer)
+    }
+}
+
+// MARK: - SCNSceneRendererDelegate (via ARSCNViewDelegate)
+extension ArcGISARView: SCNSceneRendererDelegate {
+
+    public  func renderer(_ renderer: SCNSceneRenderer, updateAtTime time: TimeInterval) {
+        arSCNViewDelegate?.renderer?(renderer, updateAtTime: time)
+    }
+
+    public func renderer(_ renderer: SCNSceneRenderer, didApplyAnimationsAtTime time: TimeInterval) {
+        arSCNViewDelegate?.renderer?(renderer, didApplyConstraintsAtTime: time)
+    }
+    
+    public func renderer(_ renderer: SCNSceneRenderer, didSimulatePhysicsAtTime time: TimeInterval) {
+        arSCNViewDelegate?.renderer?(renderer, didSimulatePhysicsAtTime: time)
+    }
+    
+    public func renderer(_ renderer: SCNSceneRenderer, didApplyConstraintsAtTime time: TimeInterval) {
+        arSCNViewDelegate?.renderer?(renderer, didApplyConstraintsAtTime: time)
+    }
+    
+    public func renderer(_ renderer: SCNSceneRenderer, willRenderScene scene: SCNScene, atTime time: TimeInterval) {
+        
+        // Get transform from SCNView.pointOfView.
+        guard let transform = arSCNView.pointOfView?.transform else { return }
+        let cameraTransform = simd_double4x4(transform)
+        
+        let finalQuat:simd_quatd = simd_mul(simd_mul(compensationQuat, simd_quatd(cameraTransform)), orientationQuat)
+        var transformationMatrix = AGSTransformationMatrix(quaternionX: finalQuat.vector.x,
+                                                           quaternionY: finalQuat.vector.y,
+                                                           quaternionZ: finalQuat.vector.z,
+                                                           quaternionW: finalQuat.vector.w,
+                                                           translationX: (cameraTransform.columns.3.x) * translationTransformationFactor,
+                                                           translationY: (-cameraTransform.columns.3.z) * translationTransformationFactor,
+                                                           translationZ: (cameraTransform.columns.3.y) * translationTransformationFactor)
+        
+        transformationMatrix = initialTransformationMatrix.addTransformation(transformationMatrix)
+        
+        let camera = AGSCamera(transformationMatrix: transformationMatrix)
+        sceneView.setViewpointCamera(camera)
+        
+        sceneView.renderFrame()
+
+        // Calculate frame rate.
+        let frametime = time - lastUpdateTime
+        print("Frame rate = \(String(reflecting: Int((1.0 / frametime).rounded())))")
```

Comment:
this should result in thousands of print statements to the console. Not sure that is ideal

Reply:
Commented it out.